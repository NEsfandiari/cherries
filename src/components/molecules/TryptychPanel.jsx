import React, { Component } from 'react'
import styled from 'styled-components'

const Container = styled.div`
  text-align: center;
  width: 25rem;
  font-size: 1rem;
  font-weight: 900;
  padding: 2rem;
  .tryptych-panel-image {
    height: 5rem;
    margin: 0 auto;
  }

  flex-basis: 33%;
  display: flex;
  flex-direction: column;
  align-items: center;
  
`

class TryptychPanel extends Component {
  render() {
    return (
      <Container>
        <div className="tryptych-panel">
          <img
            src={this.props.imageUrl}
            alt={this.props.imageName}
            className="tryptych-panel-image"
          />
          <p className="tryptych-panel-text">{this.props.text}</p>
        </div>
      </Container>
    )
  }
}

export default TryptychPanel
