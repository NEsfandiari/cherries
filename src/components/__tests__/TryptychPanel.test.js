import React from 'react'
import renderer from 'react-test-renderer'
import TryptychPanel from '../molecules/TryptychPanel'

describe('TryptychPanel', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<TryptychPanel />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
