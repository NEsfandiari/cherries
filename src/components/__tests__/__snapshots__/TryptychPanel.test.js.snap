// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TryptychPanel renders correctly 1`] = `
<div
  className="sc-bdVaJa ftRuNj"
>
  <div
    className="tryptych-panel"
  >
    <img
      className="tryptych-panel-image"
    />
    <p
      className="tryptych-panel-text"
    />
  </div>
</div>
`;
